notifier/orb.yaml

```diff
@@ -1,12 +1,56 @@
 version: 2.1
 description: "Tools for notifying external services about CI events"
 
+orbs:
+  slack: circleci/slack@4.7.0
+
 executors:
   alpine-stable:
     docker:
     - image: alpine:latest
     resource_class: small
 
+commands:
+  slack-on-deploy:
+    description: |
+      Uses slack orb to send a notification to slack when a deployment fails.
+    parameters:
+      channel:
+        default: ${SLACK_DEFAULT_CHANNEL}
+        description: |
+          Destination channel for your slack messages.
+        type: string
+    steps:
+      - slack/notify:
+          channel: <<parameters.channel>>
```

Comment:
Random thought: does this field support usernames? Just thinking of potential future improvements rather than any immediate changes, eg. that we could directly message these alerts to individual folks in cases where it's not a production job. Could even do this as a fallback (eg. try to send to `@<username>` and, assuming the step fails if that user doesn't exist, send to the channel on-fail).

Reply:
Hmm yeah, that wouldn't be ideal for a public orb. Maybe there's some way we could store that in the environment; eg. have everyone set an environment variable like `CIRCLECI_USER_TO_SLACK_leanaha = "lean"` or something along those lines. The the orb could do the equivalent of `username = os.environ.get(f'CIRCLECI_USER_TO_SLACK_{circleci_username}') or circleci_username`

 Even that wouldn't really be the easiest to work with, I imagine...

Reply:
Yeah, well, Idk how to continue with this. Should we address this part of the solution before merging to master or we can leave it behind as a tech debt? However, it would be great if we can keep this conversation somewhere and we can come back to this feature later.

Reply:
I think leaving this as-is for now is 100% OK. Our PR conversations never get deleted, so we'll be able to go back to this discussion once we want to tackle this improvement :+1: